frontend/src/static/js/main.js

```diff
@@ -181,6 +181,10 @@ window.app = new window.Vue({
       this.activateTab('authorship');
     },
     updateTabZoom(obj) {
+      // remove hashes from authorship tab if available
+      window.removeHash('tabAuthor');
+      window.removeHash('tabRepo');
```

Comment:
what is the tab zoom's role in this if its not being restored?

Reply:
This basically removes the `tabAuthor` and `tabRepo` hash that belongs to the authorship tab. Since we are switching to the zoom tab, those hashes are unnecessary.

Reply:
Aren't the details in zoom tab still tied to the author and repo?
If we are going to make zoom tab restore-able in the future, don't we need these details?

Reply:
Yes we do require those details, but if we are not removing the `tabAuthor` and `tabRepo` hash, reloading the page with zoom tab open will result to the authorship tab opened with files related to the unremoved `tabAuthor ` and `tabRepo` hash. 

![zoomtab](https://user-images.githubusercontent.com/32864116/60315587-af345800-9999-11e9-8396-26196a0a8b04.gif)

Since this PR is not about restoring the zoom tab, I can make another PR that handles the `tabAuthor` and `tabRepo` correctly and removing these two lines.
https://github.com/reposense/RepoSense/blob/2347cec8c87149eaa6389e6835456c1c9e52e91a/frontend/src/static/js/main.js#L183-L186

Reply:
>Yes we do require those details, but if we are not removing the tabAuthor and tabRepo hash, reloading the page with zoom tab open will result to the authorship tab opened with files related to the unremoved tabAuthor and tabRepo hash.

In that case, would adding the previous `tabType` (while retaining the `tabAuthor` and `tabRepo`), helps?

Reason is because it is never good to remove things that are just going to be added back in another PR. In normal circumstances, reverting such PRs wouldn't be possible without manual intervention. Hence, as a practice, we should try to avoid doing such things. This also serves as the justification for your addition of the `tabType` component.

Reply:
> In that case, would adding the previous `tabType` (while retaining the `tabAuthor` and `tabRepo`), helps?

Yes I think it'll help. Reloading the page with zoom tab will probably lead to an empty tab.

> Reason is because it is never good to remove things that are just going to be added back in another PR. 

Noted with thanks. Will try to implement it.

Reply:
@eugenepeh @fzdy1914 

I am changing the restoring of tabs to be within `v_summary` component instead of `app` component. My reasoning behind this is because:

- `v_summary` component is originally responsible of opening tabs, so it makes sense for `v_summary` to handle the restoring of tabs as well, instead of adding code in `app` component to handle restoring of tabs.
- restoring zoom tab will require `filtered` within `v_summary`, so if we handle the restoring of zoom tab within `app` component, we will not be able to access `filtered` as `v_summary` will not be mounted yet.

Additionally, I am changing the url hash from using `tabRepo, tabAuthor` to `repoIndex, userIndex`. This is because to restore zoom tab, we would require to access `filtered` within `v_summary`. The `repoIndex` and `userIndex` is used to keep track of which specific element within `filtered`.

Reply:
>Additionally, I am changing the url hash from using tabRepo, tabAuthor to repoIndex, userIndex. This is because to restore zoom tab, we would require to access the filtered array within v_summary as it consists of the repositories that are filtered based on the selected granularity. The repoIndex and userIndex is used to keep track of which specific element within filtered.

Wouldn't that requires the filtered to be loaded first, e.g. search value takes effect beforehand, for this to work?

Reply:
Yes. The filtered will be loaded first, then the tabs will be restored.

Reply:
>Yes. The filtered will be loaded first, then the tabs will be restored.

You will need to rethink about this. As your index works on the filtered, modifying the search bar will mess the filtered repos, resulting the restoring not working as expected.

Reply:
Hmmm that is true. 

But for zoom tab we will definitely require to use `filtered` as `filtered` consists of the commits being grouped in terms of the granularity. If we really need to use `repos`, then we would have to specifically group the commits in `window.REPOS` in terms of the granularity then restore the zoom tab (as opposed to just using `filtered`).

Another reason why I used `filtered` is to take into account of restoring zoom tab from merged groups in PR #662. When the merge group option is checked, `filtered` consist of the merged groups.

Reply:
You might need to figure a way out to get this working. The reason I brought it up is because the problem exist in your current work.

Reply:
I have reverted the commits and will deal with restoring of zoom tab in another PR.

Reply:
@eugenepeh I have figured out a new way of restoring zoom tab that uses `window.REPOS` instead of `filtered`. Will proceed with that after this PR.